steam_community_market/market.py

```diff
@@ -274,60 +245,117 @@ def get_volume(self, name: str, app_id: int | AppID) -> int | None:
 
         return int(item["volume"].replace(",", "")) if "volume" in item else None
 
-    def has_invalid_name(self, name: str) -> bool:
-        """Checks if given item name is invalid.
+    @staticmethod
+    def _app_id_value(
+        app_id: Union[int, list[Union[int, AppID]], AppID], support_lists: bool = False
+    ) -> int:
+        """Validates and returns the value of an AppID.
 
-        :param name: The name of the item how it appears on the Steam Community Market.
-        :type name: :class:`str`
-        :raises TypeError: If the given name is not a string.
-        :return: :class:`True` if the item name is invalid, :class:`False` otherwise.
-        :rtype: :class:`bool`
+        .. versionadded:: 1.3.0
 
-        .. versionadded:: 1.1.0
+        :param app_id: The App ID of the game the item is from.
+        :type app_id: int or list[int or AppID] or AppID
+        :param support_lists: Whether or not to support lists.
+        :type support_lists: bool
+        :raises TypeError: Raised when :param:`app_id` is not a supported type.
+        :return: The value of the App ID.
+        :rtype: int
         """
 
-        if isinstance(name, str):
-            try:
-                return name.index("/") >= 0
-            except ValueError:
-                return False
+        if isinstance(app_id, AppID):
+            return app_id.value
+        if isinstance(app_id, int) or (
+            support_lists and isinstance(app_id, list[Union[int, AppID]])

```

Comment:
isinstance checks don't work with GenericAliases, this will fail at runtime. Runtime typechecking like this is a lost cause

Reply:
Indeed, I'll be fixing this shortly.